lms/djangoapps/teams/csv.py

```diff
@@ -406,21 +432,31 @@ def is_FERPA_bubble_breached(self, teamset_id, team_name):
         else:
             return True
 
-    def validate_proposed_team_size_wont_exceed_maximum(self, team_name, teamset_id):
+    def validate_team_sizes_not_exceeded(self):
```

Comment:
I am on the fence about this approach.
Pro: It becomes a self contained function
Con: We will once again, be looping over the team sets.

Reply:
Do we see courses with a large number of team sets in the wild?

Reply:
Yeah, I was conflicted about this. This whole system is very "row-centric" that we operate on a row and then operate on the next one. To have a post-process step while using the data we already have access to requires another loop over the rows. I had considered changing the internal data structures to capture enough data that we wouldn't need to loop again but this felt like a good compromise between existing code re-use and needed functionality.

Reply:
Yep it's tough. The input doc`learner-centric`, however we have to validate on teams / teamsets that aren't.